File: scripts/generate-key.js
// @ts-check
import crypto from "crypto"
import fs from "fs"
import path from "path"

const rawToken = Buffer.from(crypto.randomBytes(48)).toString("hex")
const tokenToFeed = `om_${rawToken}`

const __pwd = process.cwd()

const envFilePath = path.join(__pwd, ".env")

if (!fs.existsSync(envFilePath)) {
  fs.writeFileSync(envFilePath, "")
}

const envFile = fs.readFileSync(envFilePath, "utf8")

const newEnvFileContent = envFile.includes("API_KEY")
  ? envFile.replace(/API_KEY=.*/, `API_KEY=${tokenToFeed}`)
  : `${envFile}\API_KEY=${tokenToFeed}`

fs.writeFileSync(envFilePath, newEnvFileContent.trim())

console.log(`Token generated! This token has already populated from an .env file. \n\n${tokenToFeed}\n`)

File: server/generate_key.go
package main

import (
	"crypto/rand"
	"encoding/hex"
	"fmt"
	"os"
	"path/filepath"
	"regexp"
)

func main() {
	bytes := make([]byte, 48)
	_, err := rand.Read(bytes)

	tokenToFeed := "om_" + hex.EncodeToString(bytes)

	cwd, err := os.Getwd()
	if err != nil {
		fmt.Println("Error getting current working directory:", err)
		return
	}

	envFilePath := filepath.Join(cwd, "..", ".env")

	var envFileContent string

	if data, err := os.ReadFile(envFilePath); err == nil {
		envFileContent = string(data)
	} else if !os.IsNotExist(err) {
		fmt.Println("Error reading .env file:", err)
		return
	}

	re := regexp.MustCompile(`(?m)^API_KEY=.*$`)

	if re.MatchString(envFileContent) {
		envFileContent = re.ReplaceAllString(envFileContent, "API_KEY="+tokenToFeed)
	} else {
		if len(envFileContent) > 0 {
			envFileContent += "\n"
		}
		envFileContent += "API_KEY=" + tokenToFeed
	}

	err = os.WriteFile(envFilePath, []byte(envFileContent), 0644)
	if err != nil {
		fmt.Println("Error writing to .env file:", err)
		return
	}

	fmt.Printf("Token generated! This token has already populated from an .env file.\n\n%s\n", tokenToFeed)
}
